ddtrace/internal/ci_visibility/git_client.py

```diff
@@ -0,0 +1,198 @@
+            yield path
+
+    @classmethod
+    def _upload_packfiles(cls, base_url, repo_url, packfiles_path, serde, _response, cwd=None):
+        # type: (str, str, str, CIVisibilityGitClientSerDeV1, Optional[Response], Optional[str]) -> bool
+        sha = extract_commit_sha(cwd=cwd)
+        parts = packfiles_path.split("/")
+        directory = "/".join(parts[:-1])
+        rand = parts[-1]
+        for filename in os.listdir(directory):
+            if not filename.startswith(rand):
+                continue
+            file_path = os.path.join(directory, filename)
+            content_type, payload = serde.upload_packfile_encode(repo_url, sha, file_path)
+            headers = {"Content-Type": content_type}
+            response = _response or cls.retry_request()(base_url, "/packfile", payload, serde, headers=headers)
+            return response.status == 204
+        return False
+
+    @classmethod
+    def retry_request(cls):
+        # type: () -> tenacity.Retrying
+        return tenacity.Retrying(
+            wait=tenacity.wait_random_exponential(
+                multiplier=(0.618 / (1.618 ** cls.RETRY_ATTEMPTS) / 2),
+                exp_base=1.618,
+            ),
+            stop=tenacity.stop_after_attempt(cls.RETRY_ATTEMPTS),
+            retry=tenacity.retry_if_exception_type((compat.httplib.HTTPException, OSError, IOError)),
+        )
+
+
+class CIVisibilityGitClientSerDeV1(object):
+    def __init__(self, api_key, app_key):
+        # type: (str, str) -> None
+        self.api_key = api_key
+        self.app_key = app_key
+
+    def search_commits_encode(self, repo_url, latest_commits):
+        # type: (str, list[str]) -> str
+        return json.dumps(
+            {"meta": {"repository_url": repo_url}, "data": [{"id": sha, "type": "commit"} for sha in latest_commits]}
+        )
+
+    def search_commits_decode(self, payload):
+        # type: (str) -> list[str]
+        parsed = json.loads(payload)
+        return [item["id"] for item in parsed["data"] if item["type"] == "commit"]
+
+    def upload_packfile_encode(self, repo_url, sha, file_path):
+        # type: (str, str, str) -> Tuple[str, bytes]
+        BOUNDARY = b"----------boundary------"
+        CRLF = b"\r\n"
+        body = []
+        metadata = {"data": {"id": sha, "type": "commit"}, "meta": {"repository_url": repo_url}}
+        body.extend(
+            [
+                b"--" + BOUNDARY,
+                b'Content-Disposition: form-data; name="pushedSha"',
+                b"Content-Type: application/json",
+                b"",
+                json.dumps(metadata).encode("utf-8"),
+            ]
+        )
+        file_name = os.path.basename(file_path)
+        f = open(file_path, "rb")
+        file_content = f.read()
+        f.close()
+        body.extend(
+            [
+                b"--" + BOUNDARY,
+                b'Content-Disposition: form-data; name="file"; filename="%s"' % file_name.encode("utf-8"),
```

Comment:
not sure that it matters, but this is `packfile` for other tracers 
```suggestion
                b'Content-Disposition: form-data; name="packfile"; filename="%s"' % file_name.encode("utf-8"),
```